### PR TITLE
Enter VideoSpace

### PR DESCRIPTION
**Pull request by smadha**
Modules added -
1. `videospace` - Forked from `imagespace` and edited for video Support
2. `videospace_tika` - Meta data based similarity plugin. Uses similarity score from https://github.com/chrismattmann/tika-similarity
3. `videospace_pot` - Pixel level time series similarity plugin based on https://github.com/USCDataScience/hadoop-pot

### CODE REVIEW

**Comment by danlamanna** [comment]
Not sure what is going on with your commit history, but can you rebase/squash or something?As it is it looks like merging this would blame every single line in the codebase to you.

**Comment by smadha** [comment]
@danlamanna Are you talking about files which were copied without changes from `imagespace` to `videospace`? Or is it happening for all files in `imagespace` too? If it's happening for all files I can do a rebase and raise another PR.

File: videospace_pot/server/__init__.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+###############################################################################
```
Comment (chrismattmann) [inline videospace_pot/server/__init__.py:4]: please remove all code comments that say copyright Kitware. Videospace should have an ALv2 header, there is no need to include the copyright Kitware @smadha

File: videospace/index/flann_index.py [hunk 2]
```diff
@@ -0,0 +1,81 @@
+import cv2
```
Comment (chrismattmann) [inline videospace/index/flann_index.py]: do we need this file? I don't think so.

File: videospace/index/lists.txt [hunk 3]
```diff
@@ -0,0 +1 @@
+listing.txt
```
Comment (chrismattmann) [inline videospace/index/lists.txt]: please remove the listing files (and listing txt, etc.) no need for them.

File: videospace/server/settings.py [hunk 4]
```diff
@@ -0,0 +1,30 @@
+import os
```
Comment (chrismattmann) [inline videospace/server/settings.py:20]: ALv2 header

**Review by chrismattmann — COMMENTED** [review]
@smadha looking good, see my comments.

File: videospace_pot/server/__init__.py (hunk 1 above)
Comment (smadha) [reply videospace_pot/server/__init__.py:4]: Do you also want me to remove copyright Kitware from all files which are copied from imagespace?

File: videospace/server/__init__.py [hunk 5]
```diff
@@ -0,0 +1,245 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+###############################################################################
+#  Copyright Kitware Inc.
```
Comment (chrismattmann) [inline videospace/server/__init__.py]: remove copyright

File: videospace/server/imagefeatures_rest.py [hunk 6]
```diff
@@ -0,0 +1,77 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+###############################################################################
+#  Copyright Kitware Inc.
```
Comment (chrismattmann) [inline videospace/server/imagefeatures_rest.py]: remove this file - is it needed?

File: videospace/server/__init__.py (hunk 5 above)
Comment (chrismattmann) [reply videospace/server/__init__.py]: this file - is it needed?

File: videospace/server/imagepivot_rest.py [hunk 7]
```diff
@@ -0,0 +1,41 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+###############################################################################
+#  Copyright Kitware Inc.
```
Comment (chrismattmann) [inline videospace/server/imagepivot_rest.py]: this file isn't needed right?

File: videospace/server/imageprefix_rest.py [hunk 8]
```diff
@@ -0,0 +1,43 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+###############################################################################
+#  Copyright Kitware Inc.
```
Comment (chrismattmann) [inline videospace/server/imageprefix_rest.py]: this file is needed?

**Review by chrismattmann — COMMENTED** [review]
more comments

**Comment by chrismattmann** [comment]
@smadha FYI my comments thanks.

**Comment by chrismattmann** [comment]
LGTM as a first cut. If there are further updates, let's open PRs against new ones. Thanks @smadha and thanks for the review @danlamanna ! Great work, Video Space is born!

**Pull request merged**